File: ios/AgentChat/Models/Conversation.swift
import Foundation

struct Conversation: Identifiable, Codable {
    let id: UUID
    let agentId: String
    var messages: [Message]
    let createdAt: Date
    var updatedAt: Date
    var sessionSuffix: String

    init(id: UUID = UUID(), agentId: String, messages: [Message] = [], createdAt: Date = Date(), updatedAt: Date = Date(), sessionSuffix: String = UUID().uuidString) {
        self.id = id
        self.agentId = agentId
        self.messages = messages
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.sessionSuffix = sessionSuffix
    }

    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        id = try container.decode(UUID.self, forKey: .id)
        agentId = try container.decode(String.self, forKey: .agentId)
        messages = try container.decode([Message].self, forKey: .messages)
        createdAt = try container.decode(Date.self, forKey: .createdAt)
        updatedAt = try container.decode(Date.self, forKey: .updatedAt)
        sessionSuffix = try container.decodeIfPresent(String.self, forKey: .sessionSuffix) ?? UUID().uuidString
    }

    /// Gateway session key — changes when the conversation is cleared.
    var sessionKey: String {
        "agent:\(agentId):ios-\(sessionSuffix)"
    }

    var lastMessage: Message? {
        messages.last { $0.role != .system }
    }
}

File: mobile/src/types/conversation.ts
import { Message } from './message';

export interface Conversation {
  id: string;           // UUID string
  agentId: string;
  messages: Message[];
  createdAt: string;    // ISO 8601
  updatedAt: string;    // ISO 8601
  sessionSuffix: string;
}

export function sessionKey(c: Conversation): string {
  return `agent:${c.agentId}:rn-${c.sessionSuffix}`;
}

export function lastMessage(c: Conversation): Message | undefined {
  for (let i = c.messages.length - 1; i >= 0; i--) {
    if (c.messages[i].role !== 'system') return c.messages[i];
  }
  return undefined;
}
